src/main/java/com/boot/product/dto/CharacteristicsDTO.java

```diff
@@ -0,0 +1,34 @@
+package com.boot.product.dto;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.Accessors;
+
+
+@Data
+@Accessors(chain = true)
+@NoArgsConstructor
+@AllArgsConstructor
+public class CharacteristicsDTO  {
+
+    /**
+     *
+     */
+    private static final long serialVersionUID = 3L;
+
+    private Long id;
+
+    private String shape;
```

Comment:
add also validation to the inputs

Reply:
you should add also the NotNull constraint. I think that this applies only if the input is not null.